.github/workflows/check-build-test.yml

```diff
@@ -62,17 +71,83 @@ jobs:
           name: ${{ github.job }}
           path: ${{ github.workspace }}/ccache-debug
 
+  sonar:
+    name: "Static Analysis"
+    runs-on: ubuntu-latest
+    container: docker.io/ogdf/clang:15
+    env:
+      CCACHE_COMPILERCHECK: "%compiler% -v"
+    steps:
+      - name: "Add workspace as a safe directory in containers"
+        run: git config --system --add safe.directory $GITHUB_WORKSPACE
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # fetch full history for blame info
+      - name: Install tools
+        run: |
+          apt-get update
+          apt-get install -y curl gcovr
```

Comment:
The Docker file could be updated to already include these, I'll do that in a later PR.

Reply:
I'll leave it in then until the Docker images are updated